lib/rucio/core/transfer.py

```diff
@@ -1061,7 +1061,9 @@ def protocol(self, rse_id, scheme, operation):
                 if current_source_is_tape and new_source_is_tape:
                     # multiple Tape source replicas are not allowed in FTS3.
                     # Either keep the old source. Or substitute it with the new one.
-                    if ranking > transfers[req_id]['sources'][0][3] or (ranking == transfers[req_id]['sources'][0][3] and link_ranking < transfers[req_id]['sources'][0][4]):
+                    if ranking > transfers[req_id]['sources'][0][3] or \
+                            (ranking == transfers[req_id]['sources'][0][3] and (not transfers[req_id]['sources'][0][4]  # will be None if the previous transfer is multihop
```

Comment:
to make this easier to comprehend, add a variable like this:

```python
prev_is_multihop = transfers[req_id]['sources'][0][4] is None  # will be None if the previous transfer is multihop
```

Reply:
Good idea. Done.